### PR TITLE
This commit refactors the controller, model and some test for challen…

### PR DESCRIPTION
**Pull request by mjj677**
…ge 07 based on John's feedback

This commit refactors the controller and model for the
POST/API/ARTICLES/:ARTICLE_ID/COMMENTS endpoint

I have removed some unnecessary tests for edge cases that psql would actually handle for me.

I have refactored the controller to use a promise.all like I did in the previous challenge (06) refactor.
I have also moved the error handling out of the postComment model and into the check functions and
catch block in the controller.

### CODE REVIEW

File: controllers/article-controllers/article-controllers.js [hunk 1]
```diff
@@ -26,25 +28,54 @@ exports.getArticles = (req, res, next) => {
...
+      res.status(200).send({ article_comments: comments });
     })
     .catch((err) => {
-        if (err.status === 404) {
-            return next({ status:404, msg: "Can't find comments at provided article"})
-        }
-        next(err)
-    })
+      if (err.status === 404) {
+        return next({
```
Comment (LangaLee) [inline controllers/article-controllers/article-controllers.js:37]: could we make this use of the next middleware and not have to specify the error in this line? Maybe we throw the error in our model ?

File: controllers/article-controllers/article-controllers.js [hunk 2]
```diff
@@ -26,25 +28,54 @@ exports.getArticles = (req, res, next) => {
...
+exports.postCommentByArticleID = (req, res, next) => {
   const { body } = req;
   const { article_id } = req.params;
-  
-  postComment(body, article_id)
+
+  const articleID = parseInt(article_id);
+
+  if (isNaN(articleID)) {
+    return next({
```
Comment (LangaLee) [inline controllers/article-controllers/article-controllers.js:53]: Could we handle this using the next middleware ?

**Review by LangaLee — COMMENTED** [review]
Great job so far, have a look at the comments below. You are doing most of you error handling in your models by rejecting errors there so lets keep it consistent and not invoke next in our controllers with error messages, lets have these errors be handled in our models and use the next middleware in a dynamic way whereby it handles whatever error is thrown 😄
